docs/Advanced-Usage/Chip-Communication.rst

```diff
@@ -0,0 +1,147 @@
+* using the JTAG interface with OpenOCD and GDB to communicate with the design
+
+Using the Tethered Serial Interface (TSI) or the Debug Module Interface (DMI)
+-----------------------------------------------------------------------------
+
+If you are using TSI or the DMI to communicate with the target (DUT/chip), you are using
+the Front-End Server (FESVR) to facilitate communication between the host machine and the target.
+
+Primer on the Front-End Server (FESVR)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+FESVR is a C++ library that manages communication
+between a host machine and a RISC-V target. For debugging, it provides a simple API to reset,
+send messages, and load/run programs on a DUT. It also emulates peripheral devices.
+It can be incorporated with simulators (VCS, Verilator, FireSim), or used in a bringup sequence
+for a taped out chip.
+
+Specifically, FESVR uses the Host Target Interface (HTIF), a communication bus for the hardware,
+to speak with the target. HTIF is a non-standard Berkeley extension that uses a FIFO non-blocking
+interface to communicate with the target.
+
+Using the Tethered Serial Interface (TSI)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Chipyard uses the Tethered Serial Interface (TSI) to communicate with the target.
+TSI is an implementation of HTIF that is used to send commands to the
+RISC-V target. These TSI commands are simple R/W commands
+that are able to probe the DUT's memory space. In simulation, these TSI commands connect
+to a ``SimSerial`` (located in the ``generators/testchipip`` project) simulation C++
+class that is added to simulation. This ``SimSerial`` device sends the TSI command to
+the DUT which contains a ``SerialAdapter`` (located in the ``generators/testchipip``
+project) that converts the TSI commands to TileLink requests. In simulation, FESVR
+resets the DUT, and writes into memory the test program. This is currently the fastest
+mechanism to communicate with the DUT in simulation.
+
+In the case of a chip tapeout bringup, TSI commands can be sent over a custom communication
+medium to communicate with the chip. For example, some Berkeley tapeouts have a FPGA
+with a RISC-V soft-core that runs FESVR. The FESVR on the soft-core sends TSI commands
+to a TSI to TileLink converter living on the FPGA (i.e. ``SerialAdapter``). Then this converter
+sends the converted TileLink commands over a serial link to the chip.
+
+Using the Debug Module Interface (DMI)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another option to bringup the target is to use the Debug Module Interface (DMI) provided by a Debug Transfer Module (DTM) existing within the target.
+Similar to TSI, DMI is an implementation of HTIF.
+The DTM is given in the `RISC-V Debug Specification <https://riscv.org/specifications/debug-specification/>`__ and is responsible for managing communication between
+the target and whatever lives on the other side of the DMI (in this case FESVR). This is added by default
```

Comment:
If you need to add the mix-ins, then I'm not sure the term "added by default" is correct?